### ISSUE TITLE
Can't run scenario tests

### ISSUE THREAD

**Issue by mboersma** [Azure/azure-cli#5366]
A recent commit to `dev` seems to have broken my ability to run scenario tests. I created a fresh virtualenv and ran `dev_setup.py` again to make sure, but all invocations of `run_tests` simply return with the `az` help message highlighted in red as if it were in error. `azdev test` acts similarly.

Please let me know if you need help reproducing this. I have some improvements to `az aks` tests that are blocked by this.

**Comment by derekbekoe** [comment]
@troydai Possibly due to this?
https://github.com/Azure/azure-cli/pull/5345
Also, is it separate from https://github.com/Azure/azure-cli/issues/5350?

**Comment by mboersma** [comment]
> is it separate from #5350?

Yes, this is on macOS with `bash` and doesn't cause a stack trace. Here's an example (I picked `postgres` randomly but it fails regardless of whether `--module` is supplied).

```
$ run_tests --module postgres
The run_tests command is going to be replaced by 'azdev test' command.

The tests are set to run against current profile /\
    /  \    _____   _ _  ___ _
   / /\ \  |_  / | | | \'__/ _\
  / ____ \  / /| |_| | | |  __/
 /_/    \_\/___|\__,_|_|  \___|


Welcome to the cool new Azure CLI!

Here are the base commands:

    account          : Manage Azure subscription information.
    acr              : Manage Azure Container Registries.
    acs              : Manage Azure Container Services.
    ad               : Synchronize on-premises directories and manage Azure Active Directory
                       resources.
    advisor          : (PREVIEW) Manage Azure Advisor.
    aks              : Manage Azure Kubernetes Services.
    appservice       : Manage App Service plans.
    backup           : Commands to manage Azure Backups.
    batch            : Manage Azure Batch.
    batchai          : Batch AI.
    billing          : Manage Azure Billing.
    cdn              : Manage Azure Content Delivery Networks (CDNs).
    cloud            : Manage registered Azure clouds.
    cognitiveservices: Manage Azure Cognitive Services accounts.
    configure        : Display and manage the Azure CLI 2.0 configuration. This command is
                       interactive.
    consumption      : Manage consumption of Azure resources.
    container        : (PREVIEW) Manage Azure Container Instances.
    cosmosdb         : Manage Azure Cosmos DB database accounts.
    disk             : Manage Azure Managed Disks.
    dla              : (PREVIEW) Manage Data Lake Analytics accounts, jobs, and catalogs.
    dls              : (PREVIEW) Manage Data Lake Store accounts and filesystems.
    eventgrid        : Manage Azure Event Grid topics and subscriptions.
    extension        : Manage and update CLI extensions.
    feature          : Manage resource provider features.
    feedback         : Loving or hating the CLI?  Let us know!
    find             : Find Azure CLI commands.
    functionapp      : Manage function apps.
    group            : Manage resource groups and template deployments.
    identity         : Managed Service Identities.
    image            : Manage custom virtual machine images.
    interactive      : Start interactive mode.
    iot              : (PREVIEW) Manage Internet of Things (IoT) assets.
    keyvault         : Safeguard and maintain control of keys, secrets, and certificates.
    lab              : Manage Azure DevTest Labs.
    lock             : Manage Azure locks.
    login            : Log in to Azure.
    logout           : Log out to remove access to Azure subscriptions.
    managedapp       : Manage template solutions provided and maintained by Independent Software
                       Vendors (ISVs).
    monitor          : Manage the Azure Monitor Service.
    mysql            : Manage Azure Database for MySQL servers.
    network          : Manage Azure Network resources.
    policy           : Manage resource policies.
    postgres         : Manage Azure Database for PostgreSQL servers.
    provider         : Manage resource providers.
    redis            : Access to a secure, dedicated Redis cache for your Azure applications.
    reservations     : Manage Azure Reservations.
    resource         : Manage Azure resources.
    role             : Manage user roles for access control with Azure Active Directory and service
                       principals.
    sf               : Manage and administer Azure Service Fabric clusters.
    snapshot         : Manage point-in-time copies of managed disks, native blobs, or other
                       snapshots.
    sql              : Manage Azure SQL Databases and Data Warehouses.
    storage          : Manage Azure Cloud Storage resources.
    tag              : Manage resource tags.
    vm               : Provision Linux or Windows virtual machines.
    vmss             : Manage groupings of virtual machines in an Azure Virtual Machine Scale Set
                       (VMSS).
    webapp           : Manage web apps..
ERROR: These modules do not exist: postgres.
No module is selected.
```

**Comment by williexu** [comment]
this seems to have occurred after commit 03a21635f26a95b17f4becf32d89c2179fbe8dfa
@tjprescott could you take a look?

**Comment by tjprescott** [comment]
This occurs because of the fix for #5350.  In the `tools/automation/test/__init__.py` script, if you set `shell=True` to `shell=False`, this will work.  It seems this needs to be `false` for bash but `true` for CMD.  That will let you run the tests locally until the more robust solution is merged.
